aioitertools/tests/asyncio.py

```diff
@@ -126,3 +127,30 @@ async def fn(arg, fail=False):
         )
         self.assertEqual(result[1], 0.001)
         self.assertIsInstance(result[0], MyException)
+
+    @async_test
+    async def test_gather_cancel(self):
+        cancelled = False
+        started = False
+
+        async def _fn():
+            nonlocal started, cancelled
+            try:
+                started = True
+                await asyncio.sleep(99999)  # might as well be forever
```

Comment:
I'd prefer to see a shorter number, like 10 seconds, so that tests don't just hang for a day if the cancellation stops working.